src/nvim/ex_getln.c

```diff
@@ -4272,20 +4272,33 @@ void globpath(char_u *path, char_u *file, garray_T *ga, int expand_options)
 *  Command line history stuff	 *
 *********************************/
 
-/*
- * Translate a history character to the associated type number.
- */
-static int hist_char2type(int c)
+/// Translate a history character to the associated type number
+static HistoryType hist_char2type(const int c)
+  FUNC_ATTR_CONST FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  if (c == ':')
-    return HIST_CMD;
-  if (c == '=')
-    return HIST_EXPR;
-  if (c == '@')
-    return HIST_INPUT;
-  if (c == '>')
-    return HIST_DEBUG;
-  return HIST_SEARCH;       /* must be '?' or '/' */
+  switch (c) {
+    case ':': {
+      return HIST_CMD;
+    }
+    case '=': {
+      return HIST_EXPR;
+    }
+    case '@': {
+      return HIST_INPUT;
+    }
+    case '>': {
+      return HIST_DEBUG;
+    }
+    case '/':
+    case '?': {
+      return HIST_SEARCH;
+    }
+    default: {
+      assert(false);
```

Comment:
According to some comments
`getexline()` pases NUL for `:append` to `getcmdline()` and
`getcmdline()` pases NUL for `:insert` to `command_line_enter()`.
If we use these two commands, Nvim aborts.